content/200-concepts/100-components/03-prisma-migrate/010-mental-model.mdx

```diff
@@ -0,0 +1,137 @@
+
+Prisma Migrate's [`migrate dev` command](/reference/api-reference/command-reference#migrate-dev) allows you to track the changes you make to your database. The `migrate dev` command automatically generates SQL migration files (saved in `/prisma/migrations`) and applies them to the database. When a migration is applied to the database, the migrations table (`_prisma_migrations`) in your database is also updated.
+
+![Prisma Migrate dev flow](https://user-images.githubusercontent.com/33921841/189875756-5df6ba94-c7a9-4e9c-a8f7-ce4d5d09b204.png)
+
+The `migrate dev` command tracks the database' state using the:
+
+- Prisma schema
+- Migrations history
+- Migrations table
+- Database schema
+
+You can customise migrations before you apply them to the database using the `--create-only` flag. For example, you might want to edit a migration if you want to rename columns without incurring any data loss or load database extensions (in PostgreSQL) and database views.
+
+Under the hood, Prisma Migrate uses a [shadow database](/concepts/components/prisma-migrate/shadow-database) to detect a [schema drift](/concepts/components/prisma-migrate/shadow-database#detecting-schema-drift) and generate new migrations.
+
+> **Note**: `migrate dev` is intended to be used only in development with a disposable database.
+
+If `migrate dev` detects a schema drift or a migration history conflict, you will be prompted to reset your database — drop and recreate your database — to sync the migration history and the database schema.
+
+<details>
+
+<summary> Expand to see the shadow database explained using a cartoon</summary>
+
+![A cartoon that shows how the shadow database works.](shadow-database.png)
+
+</details>
+
+#### Resolve schema drifts
+
+A schema drift might occur when the expected database schema is different from what is in the migration history. For example, this might occur if when you manually update the database schema.
+
+For such instances, you can use Prisma's [`migrate diff` command](/reference/api-reference/command-reference#migrate-diff) to compare your migration history and revert changes made to your database schema.
+
+![Revert database schema with `migrate diff`](https://user-images.githubusercontent.com/33921841/189894555-b1dedf54-a8eb-4e29-9bf4-85a5615030b4.png)
+
+You can use `migrate diff` to generate the SQL that either:
+
+- Reverts the changes made in the database schema
+- Moves your database schema forward to apply missing changes
+
+You can then apply the changes to your database using the [`db execute` command](/reference/api-reference/command-reference#db-execute).
+
+### Production and staging environments
+
+#### Sync your migration histories
+
+Prisma Migrate's [`migrate deploy` command](/reference/api-reference/command-reference#migrate-deploy) allows you to sync your migration history from your development environment with your database in your preview or production environment.
+
+Under the hood, the `migrate deploy` command:
```

Comment:
Could clarify that `migrate deploy` is run in your production/preview environment.

Reply:
This is specified in the last part of the sentence. I'll make it bold